flamedisx/x1t_sr1.py

```diff
@@ -104,4 +115,10 @@ def s2_acceptance(s2):
                         tf.zeros_like(s2, dtype=fd.float_type()),
                         tf.ones_like(s2, dtype=fd.float_type()))
 
-#TODO: add NRsource
+class SR1NRSource(SR1Source, fd.NRSource):
+    extra_needed_columns = tuple(set(
+        list(SR1Source.extra_needed_columns) + 
+        list(fd.NRSource.extra_needed_columns)+
+        ['x_observed', 'y_observed']))
```

Comment:
I think `['x_observed', 'y_observed']` here is redundant since it's already in `SR1Source.extra_needed_columns`.